src/Symfony/Component/Lock/Store/RedisStore.php

```diff
@@ -284,8 +287,18 @@ private function evaluate(string $script, string $resource, array $args): mixed
 
         \assert($this->redis instanceof \Predis\ClientInterface);
 
-        $result = $this->redis->evalSha($scriptSha, 1, $resource, ...$args);
-        if ($result instanceof Error && self::NO_SCRIPT_ERROR_MESSAGE === $result->getMessage()) {
+        try {
+            $result = $this->redis->evalSha($scriptSha, 1, $resource, ...$args);
+        } catch (ServerException $exception) {
```

Comment:
I don't think this change makes sense: the predis factory always disables exceptions.
It's true nothing prevents another factory to be used, but this would still remain an unrelated change.

Reply:
Well, I dumped there to check what's happening and I got exactly `ServerException` for the Valkey. Most likely some other stuff makes `predis` work like that under the hood...

Reply:
how did you create the connection object?

Reply:
@nicolas-grekas snc_redis bundle by default throws exceptions

https://github.com/snc/SncRedisBundle/blob/1577b80bda8b488c45ebe14fdb72e42273c3aa8b/src/DependencyInjection/Configuration/Configuration.php#L131C52-L131C64

Reply:
The evalSha call here

https://github.com/symfony/symfony/blob/5876c48757182da02cbea275a40059b5bfc83854/src/Symfony/Component/Lock/Store/RedisStore.php#L287

throws ServerError to snc redis users with default config.

I had to add option settings:

```yaml
snc_redis:
  clients:
    locks:
      type: predis
      alias: locks
      dsn: '%redis.dsn%/0'
      logging: '%kernel.debug%'
      options:
        throw_errors: false
```